src/libraries/Native/Unix/System.Native/pal_random.c

```diff
@@ -55,6 +55,10 @@ void SystemNative_GetNonCryptographicallySecureRandomBytes(uint8_t* buffer, int3
 #endif // HAVE_ARC4RANDOM_BUF
 }
 
+#ifdef __EMSCRIPTEN__
+int32_t mono_wasm_browser_crypto_getRandomValues(uint8_t* buffer, int32_t bufferLength);
```

Comment:
Can the implementation be here instead of in mono runtime?

Reply:
The code that inlines the javascript code does not generate correctly from the dotnet tree.

```
EM_JS(int32_t, mono_wasm_browser_crypto_getRandomValues, (uint8_t* buffer, int32_t bufferLength),
 {
... inline javascript code
}
```

Either way the javascript code will end up being in one of the javascript libraries that this will call as well as atjhe definition to define it. 

The 

```
int32_t mono_wasm_browser_crypto_getRandomValues(uint8_t* buffer, int32_t bufferLength);
```

Will end up being something like the following anyway:

```
EMSCRIPTEN_KEEPALIVE int32_t 
mono_wasm_browser_crypto_getRandomValues(uint8_t* buffer, int32_t bufferLength);
```

And the code actually moved into the javascript libraries.  

The implantation here is actually clearer in my opinion.  Either way let me know if one prefers the above instead.

Reply:
> The code that inlines the javascript code does not generate correctly from the dotnet tree.

What is the problem? Should we make it work correctly instead?

The System.*.Native shims are meant to be standalone and not depend on the runtime. The shims have no other calls to the runtime today.

Reply:
It is the javascript code is all.  It needs to be in the javascript libraries so I will move it to there and there will be no reference to the runtime but will be linked with the javascript code implementation itself.

Reply:
Ok, referencing it that way will not work either.  Not sure how we will get it without referencing that way.

Reply:
Both pal_random.c  and driver.c are C files. Why does the javascript work in driver.c and not in pal_random.c ?

Is it because of the build steps for each are different? If yes, can we make the build steps for pal_random.c same as the build steps for driver.c so that the javascript works in pal_random.c as well?

Reply:
> Ok, referencing it that way will not work either. Not sure how we will get it without the reference the way it is.

What exactly are you trying and how is it failing?

Reply:
This no longer inlines the JS in driver.c.